Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+HTTP1StateMachine.swift

```diff
@@ -291,6 +291,15 @@ extension HTTPConnectionPool {
                 connection: .cleanupConnections(cleanupContext, isShutdown: isShutdown)
             )
         }
+        
+        mutating func migrateToHTTP2() -> HTTP2MigrationContext {
+            let connectionsMigrationContext = self.connections.migrateToHTTP2()
+            return HTTP2MigrationContext(
+                generator: connectionsMigrationContext.generator,
+                startingConnections: connectionsMigrationContext.starting,
```

Comment:
We need to hand over the `HTTP1Connections` as a whole. Then we can call the migrate methods on `HTTP1Connections` from the `HTTP2StateMachine`.